### PR TITLE
Feat/widget subscribe

### PR DESCRIPTION
**Pull request by devceline**
# Description

- Add widget subscribe page
- Add `dappHooks` to remove responsibility of `dappContext` from `uiHooks`
- Add relevant styling to header
- Move all modals out of `App`, and into their own component. (Reduces clutter and allows displaying modals on non `App` pages like the subscribe page)
- Add redirection to `/widget/subscribe` when necessary

# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/contexts/W3iContext/hooks/pushHooks.ts [hunk 1]
```diff
@@ -12,6 +13,8 @@ export const usePushState = (w3iProxy: Web3InboxProxy, proxyReady: boolean) => {
     PushClientTypes.PushSubscription[]
   >([])
 
+  const { pathname } = useLocation()
```
Comment (bkrem) [inline src/contexts/W3iContext/hooks/pushHooks.ts:16]: Needed? Don't see this being used in the diffs but maybe I'm missing something

**Review by bkrem — APPROVED** [review]
Nice, liking the reorg work that happened here too ✨

File: src/contexts/W3iContext/hooks/pushHooks.ts (hunk 1 above)
Comment (devceline) [reply src/contexts/W3iContext/hooks/pushHooks.ts:16]: it's used to allow in the dependency array in the below `useEffect`. This is to allow the modal to retrigger during navigation to `/widget/subscribe`

Comment (bkrem) [reply src/contexts/W3iContext/hooks/pushHooks.ts:16]: My bad, didn't realise it was a dep array at first 🤦‍♂️

**Review by bkrem — APPROVED** [review]

**Pull request merged**